android/app/src/main/java/com/cBreez/client/BreezSdkWorker.kt

```diff
@@ -0,0 +1,215 @@
+package com.cBreez.client
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.ForegroundInfo
+import androidx.work.WorkerParameters
+import breez_sdk.BlockingBreezServices
+import breez_sdk.BreezEvent
+import breez_sdk.EnvironmentType
+import breez_sdk.EventListener
+import breez_sdk.GreenlightNodeConfig
+import breez_sdk.NodeConfig
+import breez_sdk.Payment
+import breez_sdk.PaymentDetails
+import breez_sdk.PaymentStatus
+import breez_sdk.connect
+import breez_sdk.defaultConfig
+import breez_sdk.mnemonicToSeed
+import com.cBreez.client.Constants.NOTIFICATION_ID_PAYMENT_RECEIVED
+import io.flutter.util.PathUtils
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.Timer
+import java.util.TimerTask
+
+
+// SDK events listener
+class SDKListener(private val paymentListener: (payment: Payment) -> Job) : EventListener {
+    override fun onEvent(e: BreezEvent) {
+        Log.v(TAG, "Received event $e")
+        if (e is BreezEvent.InvoicePaid) {
+            Log.v(
+                TAG,
+                "Received payment. Bolt11: ${e.details.bolt11}\nPayment Hash:${e.details.paymentHash}"
+            )
+            e.details.payment?.let {
+                paymentListener(it)
+            }
+        }
+    }
+
+    companion object {
+        private const val TAG = "SDKListener"
+    }
+}
+
+open class BreezSdkWorker(appContext: Context, workerParams: WorkerParameters) :
+    CoroutineWorker(appContext, workerParams) {
+    private var preferences: SharedPreferences? = null
+    private var breezSDK: BlockingBreezServices? = null
+    private var receivedPayments: ArrayList<String> = ArrayList()
+    private var paymentHashPollerTimer: Timer? = null
+
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+        return ForegroundInfo(
+            NOTIFICATION_ID_PAYMENT_RECEIVED,
+            createNotification(
+                "Incoming payment",
+                NotificationManager.IMPORTANCE_LOW,
+                true
+            )
+        )
+    }
+
+    override suspend fun doWork(): Result = coroutineScope {
+        val paymentHash =
+            inputData.getString("PAYMENT_HASH") ?: return@coroutineScope Result.failure()
+        CoroutineScope(Dispatchers.Default).launch {
```

Comment:
Doesn't this mean it can run in multiple thread? We want breez SDK to be created once and this will allow multiple routines won't it?
I think on Android we can perhaps just use mutex to serialize operations?

There are two basic operations we need to serialize:
1. connect to breezSDK - which should be static instance.
2. sending notifications to paymentReceivers and removing from this list.